core/templates/cowdata.h

```diff
@@ -222,12 +222,15 @@ class CowData {
 	}
 
 	Error insert(Size p_pos, const T &p_val) {
-		ERR_FAIL_INDEX_V(p_pos, size() + 1, ERR_INVALID_PARAMETER);
-		resize(size() + 1);
-		for (Size i = (size() - 1); i > p_pos; i--) {
-			set(i, get(i - 1));
+		Size new_size = size() + 1;
+		ERR_FAIL_INDEX_V(p_pos, new_size, ERR_INVALID_PARAMETER);
+		Error err = resize(new_size);
+		ERR_FAIL_COND_V(err, err);
+		T *p = ptrw();
+		for (Size i = new_size - 1; i > p_pos; i--) {
```

Comment:
Shouldn't this be `new_size - 2`?

Reply:
Ignoring any resize errors, `new_size` is equal to `size() + 1` before the resize, and equal to `size()` after the resize.  `new_size - 1` matches the original code's `size() - 1` at this point in the function to get the last item in the array.

Reply:
Ah, right! Thank you for the explanation.